engine/text.cpp

```diff
@@ -0,0 +1,180 @@
+#include "text.h"
+
+#include "logger.h"
+
+#define STB_TRUETYPE_IMPLEMENTATION
+#include <stb_truetype.h>
+
+#include <fstream>
+
+namespace pong
+{
+
+Text::Text(const std::string& text, const std::string& path, float scale, int pixelLineHeight) :
+    mText { text },
+    mFontPath { path },
+    mScale { scale },
+    mPixelLineHeight { pixelLineHeight }
+{
+    CreateText();
+}
+
+void Text::CreateText()
```

Comment:
After writing this function. I must say, text rendering is probably my least favorite part of graphics rendering pipelines 🤣